scenario/scenario.go

```diff
@@ -313,3 +265,11 @@ func GetTestSignerFn(signerAddress common.Address) func(signer types.Signer, add
 		return tx.WithSignature(signer, signature)
 	}
 }
+
+func waitTxnSuccessfullyMined(t *testing.T, ethClient *ethclient.Client, txn *types.Transaction) {
+	ctx, cancelFn := context.WithTimeout(context.Background(), 4*time.Second)
+	receipt, err := bind.WaitMined(ctx, ethClient, txn)
+	require.NoError(t, err)
+	cancelFn()
```

Comment:
nit: I think `defer cancelFn()` immediately after creating the context is cleaner and less brittle. As written, the context might not be canceled if we ever changed the code in the future or a panic caused the function to return early.

It's especially important outside of the context of tests. Consider cases where:

1. An unexpected panic causes a function to return early. The code never reaches `cancelFn` so the context is not canceled.
2. We recover from that panic and resume program execution.
3. The context is never canceled, potentially creating a goroutine leak.

To address issues like this, it is generally better to call `defer cancelFn` immediately. (It is less important here because we are not recovering from panics and the test binary will always exit).